# This file is part of HiGFXback

# requires
REQUIRES="make-build opengl-registry-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=/dfb

# build
make

# install
install -d $DESTDIR/dfb/lib
install libEGL.so.1 $DESTDIR/dfb/lib/libstubEGL.so.1
test -f $DESTDIR/dfb/lib/libEGL.so.1 || ln -s libstubEGL.so.1 $DESTDIR/dfb/lib/libEGL.so.1
ln -sf libstubEGL.so.1 $DESTDIR/dfb/lib/libEGL.so
install libGL.so.1 $DESTDIR/dfb/lib/libstubGL.so.1
test -f $DESTDIR/dfb/lib/libGL.so.1 || ln -s libstubGL.so.1 $DESTDIR/dfb/lib/libGL.so.1
ln -sf libstubGL.so.1 $DESTDIR/dfb/lib/libGL.so
install libGLESv1_CM.so.1 $DESTDIR/dfb/lib/libstubGLESv1_CM.so.1
test -f $DESTDIR/dfb/lib/libGLESv1_CM.so.1 || ln -s libstubGLESv1_CM.so.1 $DESTDIR/dfb/lib/libGLESv1_CM.so.1
ln -sf libstubGLESv1_CM.so.1 $DESTDIR/dfb/lib/libGLESv1_CM.so
install libGLESv2.so.2 $DESTDIR/dfb/lib/libstubGLESv2.so.2
test -f $DESTDIR/dfb/lib/libGLESv2.so.2 || ln -s libstubGLESv2.so.2 $DESTDIR/dfb/lib/libGLESv2.so.2
ln -sf libstubGLESv2.so.2 $DESTDIR/dfb/lib/libGLESv2.so
install -d $DESTDIR/dfb/lib/directfb-2.0-0/interfaces/IDirectFBGL
install -d $DESTDIR/dfb/lib/pkgconfig
install -m 644 egl.pc $DESTDIR/dfb/lib/pkgconfig
install -m 644 gl.pc $DESTDIR/dfb/lib/pkgconfig
install -m 644 glesv1_cm.pc $DESTDIR/dfb/lib/pkgconfig
install -m 644 glesv2.pc $DESTDIR/dfb/lib/pkgconfig

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/egl-opengl-stubs-build.pc << EOF
Name: EGL-OpenGL-Stubs
Version: 0.5
Description: EGL and OpenGL stubs for linking at build time
Requires: $REQUIRES

devel=\\
/dfb/lib/libEGL.so \\
/dfb/lib/libGL.so \\
/dfb/lib/libGLESv1_CM.so \\
/dfb/lib/libGLESv2.so \\
/dfb/lib/pkgconfig/egl.pc \\
/dfb/lib/pkgconfig/gl.pc \\
/dfb/lib/pkgconfig/glesv1_cm.pc \\
/dfb/lib/pkgconfig/glesv2.pc

exec=\\
/dfb/lib/libEGL.so.1 \\
/dfb/lib/libGL.so.1 \\
/dfb/lib/libGLESv1_CM.so.1 \\
/dfb/lib/libGLESv2.so.2
/dfb/lib/libstubEGL.so.1 \\
/dfb/lib/libstubGL.so.1 \\
/dfb/lib/libstubGLESv1_CM.so.1 \\
/dfb/lib/libstubGLESv2.so.2
EOF

exit

# patch
--- EGL-OpenGL-Stubs-0.5.orig/Configure
+++ EGL-OpenGL-Stubs-0.5/Configure
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile
+
+sed -i "/^prefix/s|=.*|=$PREFIX|" egl.pc
+sed -i "/^prefix/s|=.*|=$PREFIX|" gl.pc
+sed -i "/^prefix/s|=.*|=$PREFIX|" glesv1_cm.pc
+sed -i "/^prefix/s|=.*|=$PREFIX|" glesv2.pc

# source
https://github.com/caramelli/EGL-OpenGL-Stubs/archive/v0.5/EGL-OpenGL-Stubs-0.5.tar.gz
